daemon/src/routes_taker.rs

```diff
@@ -195,14 +194,15 @@ pub async fn get_wallet_mnemonic(
     Ok(status::Accepted(Option::from(mnemonic.to_string())))
 }
 
-#[rocket::post("/mnemonic")]
-pub async fn post_mnemonic() -> Result<status::Accepted<String>, HttpApiProblem> {
-    let mnemonic =
-        Mnemonic::generate_in_with(&mut thread_rng(), Language::English, 24).map_err(|e| {
-            HttpApiProblem::new(StatusCode::INTERNAL_SERVER_ERROR)
-                .title("Generate new mnemonic request failed")
-                .detail(e.to_string())
-        })?;
+#[rocket::post("/wallet/reinitialize")]
+pub async fn post_wallet_reinitalize(
+    taker: &State<Taker>,
+) -> Result<status::Accepted<String>, HttpApiProblem> {
+    let mnemonic = taker.reinitialize_wallet().await.map_err(|e| {
+        HttpApiProblem::new(StatusCode::INTERNAL_SERVER_ERROR)
+            .title("Generate new mnemonic request failed")
+            .detail(e.to_string())
+    })?;
     Ok(status::Accepted(Option::from(mnemonic.to_string())))
```

Comment:
I know you've just continued the existing pattern here but technically 202 ACCEPTED is not really appropriate here. Same as with HTTP methods, if you don't want to communicate anything specific and it is successful, use 200 OK.